File: LeetCode/Java/Easy/643. Maximum Average Subarray I/Solution.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        double curr_sum = 0;
        double max = -100000000;
        int left = 0;
        for(int right=0; right<nums.length; right++){
            curr_sum+=nums[right];
            if(right >= k-1){
                double cAvg = curr_sum/k;
                max = Math.max(max, cAvg);
                curr_sum -= nums[left];
                left++;
            }
        }
        return max;
    }
}

File: LeetCode/C++/Easy/643. Maximum Average Subarray I/Solution.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        int n = nums.size();
        int sum = 0;
        for(int i=0; i<k; i++){
            sum += nums[i];
        }
        int l = 0, r = k-1;
        int maxSum = sum;
        while(r<n-1){
            sum -= nums[l];
            l++;
            r++;
            sum += nums[r];
            maxSum = max(maxSum, sum);
        }
        return (double)maxSum/k;
    }
};